### PR TITLE
Redo

### PR DESCRIPTION
**Pull request by cliffoo**
This box needs some love.

New look

Notable changes
- React
  - UI to clearly communicate what is happening (call and send tx to contract)
  - Class => function components, with various hooks (context, reducer, ...)
  - Components extensible to more complex dapp needs
  - Dependencies update
- Truffle
  - Moved project into /truffle subdir, clearer file structure
  - Config contains more info
- Initial setup is just unbox + run react dev server. Further setup instruction is on hosted react app page.
- Readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,71 +1,40 @@
...
+Start the react dev server.
 
-5. In the `client` directory, we run the React app. Smart contract changes must be manually recompiled and migrated.
-    ```javascript
-    // in another terminal (i.e. not in the truffle develop prompt)
-    cd client
-    npm run start
-    ```
+```sh
+$ cd client
```
Comment (leeftk) [inline README.md:25]: I think we need a `npm install in here` before we can do `npm run start`

File: truffle/contracts/Migrations.sol [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
-  constructor() public {
-    owner = msg.sender;
+    require(
+      msg.sender == owner,
+      "This function is restricted to the contract's owner"
+    );
+    _;
   }
 
   function setCompleted(uint completed) public restricted {
```
Comment (leeftk) [inline truffle/contracts/Migrations.sol:16]: Should we remove this file completely? Up for debate but we discussed all future boxes not including this anymore as its not really used and tends to waste gas for new users. Your call though.

File: client/src/components/Demo/ContractBtns.jsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+          value={inputValue}
+          onChange={handleInputChange}
+        />)
+      </div>
+
+    </div>
+  );
+}
+
+export default ContractBtns;
```
Comment (leeftk) [inline client/src/components/Demo/ContractBtns.jsx:51]: This is sickkkkkk. My only small concern is that do you think someone who is new to development may be confused by what's going on here? Clearly up to you to decide but just wanted to point out that the boxes are meant to be a bit more of a scaffold. Though these days they're starting to become a lot more complex which I sorta like. Anyway just a thought, no action items on this.

**Review by leeftk — COMMENTED** [review]
Oh also can we delete the `test` folder since there's nothing in it or do we want to keep it to show users the proper folder structure in case they want to write tests?

File: truffle/contracts/Migrations.sol (hunk 2 above)
Comment (cliffoo) [reply truffle/contracts/Migrations.sol:16]: Yea this should probably go. I believe @gnidan removed this along with its migration file in past Truffle demos.

File: README.md (hunk 1 above)
Comment (leeftk) [reply README.md:25]: ahhhh gotcha I didn't unbox I cloned which makes sense why I didn't work for me. Sick UX though if that's the case!

File: truffle/contracts/Migrations.sol (hunk 2 above)
Comment (cliffoo) [reply truffle/contracts/Migrations.sol:16]: And yea I want to keep the test directory to reflect a full Truffle project structure.

Comment (cliffoo) [reply truffle/contracts/Migrations.sol]: Resolved in [this commit](https://github.com/truffle-box/react-box/commit/aff17a2efc556030b0ad91ebe443e7663ba45b98)!

**Comment by cliffoo** [comment]
Ye I want to keep the test directory to reflect a full Truffle project structure, let's keep it.

**Comment by revoltez** [comment]
could you also correct the the migrations in artifacts.require("contractname") instead of artifacts.require("pathTocontract")

**Review by gnidan — APPROVED** [review]
looks good!

**Pull request merged**